bindings/cpp/session.cpp

```diff
@@ -2287,6 +2340,110 @@ async_iterator_result session::cancel_iterator(const key &id, uint64_t iterator_
 	return iterator(id, data);
 }
 
+async_iterator_result session::server_send(const std::vector<std::string> &keys, uint64_t iflags, const std::vector<int> &groups)
+{
+	if (get_groups().empty()) {
+		async_iterator_result result(*this);
+		async_result_handler<iterator_result_entry> handler(result);
+		handler.complete(create_error(-ENXIO, "server_send: local group list is empty"));
+		return result;
+	}
+
+	if (groups.empty()) {
+		async_iterator_result result(*this);
+		async_result_handler<iterator_result_entry> handler(result);
+		handler.complete(create_error(-ENXIO, "server_send: remote group list is empty"));
+		return result;
+	}
+
+	if (keys.empty()) {
+		async_iterator_result result(*this);
+		async_result_handler<iterator_result_entry> handler(result);
+		handler.complete(create_error(-ENXIO, "server_send: key list is empty"));
+		return result;
+	}
+
+	int local_group = get_groups().front();
+	int err;
+
+	struct la {
+		dnet_addr	addr;
+		int		backend_id;
+		dnet_id		id;
+
+		bool operator<(const la &other) const {
+			int cmp = dnet_addr_cmp(&addr, &other.addr);
+			if (cmp < 0)
+				return true;
+			if (cmp > 0)
+				return false;
+			return backend_id < other.backend_id;
+		}
+	};
+
+	std::map<la, std::vector<dnet_raw_id>> raw_ids;
+	for (const auto &key : keys) {
+		la l;
+
+		transform(key, l.id);
+
+		err = dnet_lookup_addr(get_native(), NULL, 0, &l.id, local_group, &l.addr, &l.backend_id);
+		if (err != 0) {
+			async_iterator_result result(*this);
+			async_result_handler<iterator_result_entry> handler(result);
+			handler.complete(create_error(-ENXIO,
```

Comment:
It is ENXIO already, but if it would be different, iterateor's err has to be consistent with 'no group' route table error, which is a common -6 error in our code